### PR TITLE
Fix `peft`-`transformers` compatibility

### PR DESCRIPTION
**Pull request by nikita-savelyevv**
# What does this PR do?

Install `peft==0.13.*` with `transformers==4.36.0`, because the newly released `peft==0.14` is not compatible with this transformers version.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by nikita-savelyevv** [comment]
Example of a failed test job: https://github.com/huggingface/optimum-intel/actions/runs/12231972181/job/34116078858?pr=1055

File: .github/workflows/test_openvino.yml [hunk 1]
```diff
@@ -50,6 +50,11 @@ jobs:
         run: |
           pip install transformers==${{ matrix.transformers-version }} accelerate==0.*
```
Comment (IlyasMoutawwakil) [inline .github/workflows/test_openvino.yml:51]:
```suggestion
          pip install transformers==${{ matrix.transformers-version }} accelerate==0.* peft==0.13.*
```

File: .github/workflows/test_openvino_slow.yml [hunk 2]
```diff
@@ -49,6 +49,11 @@ jobs:
         name: Downgrade Transformers and Accelerate
         run: pip install transformers==${{ matrix.transformers-version }} accelerate==0.*
```
Comment (IlyasMoutawwakil) [inline .github/workflows/test_openvino_slow.yml:50]:
```suggestion
        run: pip install transformers==${{ matrix.transformers-version }} accelerate==0.* peft==0.13.*
```

File: .github/workflows/test_openvino_slow.yml [hunk 3]
```diff
@@ -49,6 +49,11 @@ jobs:
...
 
+      - if: ${{ matrix.transformers-version == '4.36.0' }}
+        name: Install peft==0.13
+        run: |
+          pip install peft==0.13.*  
+
```
Comment (IlyasMoutawwakil) [inline .github/workflows/test_openvino_slow.yml:56]:
```suggestion
```

File: .github/workflows/test_openvino.yml [hunk 4]
```diff
@@ -50,6 +50,11 @@ jobs:
...
 
+      - if: ${{ matrix.transformers-version == '4.36.0' }}
+        name: Install peft==0.13
+        run: |
+          pip install peft==0.13.*  
+
```
Comment (IlyasMoutawwakil) [inline .github/workflows/test_openvino.yml:57]:
```suggestion
```

File: .github/workflows/test_openvino.yml (hunk 1 above)
Comment (nikita-savelyevv) [reply .github/workflows/test_openvino.yml]: Done

**Pull request merged**